files_test.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+)
+
+var sources = []string{
+	"LICENSE",
+	"README.md",
+	"fields.go",
+	"fields_test.go",
+	"files_test.go",
+	"main.go",
+	"parser.go",
+	"parser_test.go",
+	"tag.go",
+	"tag_test.go",
+	"tags",
+	"tests/const.go-src",
+	"tests/func.go-src",
+	"tests/import.go-src",
+	"tests/interface.go-src",
+	"tests/range.go-src",
+	"tests/simple.go-src",
+	"tests/struct.go-src",
+	"tests/type.go-src",
+	"tests/var.go-src",
+}
+
+func TestCommandLineFiles(t *testing.T) {
+	patterns := patternList{}
+
+	files, err := getFileNames(sources, false, patterns)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if !reflect.DeepEqual(sources, files) {
+		t.Errorf("%+v != %+v", sources, files)
+	}
+}
+
+func TestSingleExcludePattern(t *testing.T) {
+	// Single pattern - exclude *_test.go files
+	patterns := []string{"*_test.go"}
+	files, err := getFileNames(sources, false, patterns)
+	if err != nil {
+		t.Error(err)
+	}
+
+	for _, f := range files {
+		if strings.HasSuffix(f, "_test.go") {
+			t.Errorf("%v should not be included", f)
+		}
+	}
+}
+
+func TestMultipeExcludePatterns(t *testing.T) {
+	// Multiple patterns - exclude *_test.go and **/*-src
+	patterns := []string{"*_test.go", "**/*-src"}
+	files, err := getFileNames(sources, false, patterns)
+	if err != nil {
+		t.Error(err)
+	}
+
+	for _, f := range files {
+		if strings.HasSuffix(f, "_test.go") || strings.HasSuffix(f, "-src") {
+			t.Errorf("%v should not be included", f)
+		}
+	}
+}
+
+func TestRecursiveExcludes(t *testing.T) {
```

Comment:
This test can be removed. I don't have Go installed in that directory (the test doesn't fail in that case btw), so it's not so useful. For now, I trust the implementation of filepath.Match and the previous tests should be enough to test that it's being used correctly.